### ISSUE TITLE
Cannot install on Windows 10 with MSVS Build Tools 2019

### ISSUE THREAD

**Issue by Antillar** [maljovec/pyerg#20]
OS system: Windows 10 x64 Home
Tried 
`pip install nglpy`
Error
`error: Microsoft Visual C++ 14.0 is required. Get it with "Build Tools for Visual Studio": https://visualstudio.microsoft.com/downloads/`
Went and installed MSVS Build Tools 2019 and ticked "C++ build tools"
![image](https://user-images.githubusercontent.com/12700019/116994781-d4a2f300-ace1-11eb-891c-aa246840db45.png)
Trying to install again:
`pip install nglpy`
Got error:
`Collecting nglpy
  Using cached nglpy-1.1.2.tar.gz (78 kB)
Requirement already satisfied: scipy<2.0,>=1.3.2 in c:\users\mahir\appdata\local\programs\python\python39\lib\site-packages (from nglpy) (1.6.2)
Requirement already satisfied: numpy<2.0,>=1.17.4 in c:\users\mahir\appdata\local\programs\python\python39\lib\site-packages (from nglpy) (1.20.2)
Requirement already satisfied: scikit-learn<1.0,>=0.21.2 in c:\users\mahir\appdata\local\programs\python\python39\lib\site-packages (from nglpy) (0.24.1)
Requirement already satisfied: joblib>=0.11 in c:\users\mahir\appdata\local\programs\python\python39\lib\site-packages (from scikit-learn<1.0,>=0.21.2->nglpy) (1.0.1)
Requirement already satisfied: threadpoolctl>=2.0.0 in c:\users\mahir\appdata\local\programs\python\python39\lib\site-packages (from scikit-learn<1.0,>=0.21.2->nglpy) (2.1.0)
Building wheels for collected packages: nglpy
  Building wheel for nglpy (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'c:\users\mahir\appdata\local\programs\python\python39\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\mahir\\AppData\\Local\\Temp\\pip-install-cjdf7ksc\\nglpy\\setup.py'"'"'; __file__='"'"'C:\\Users\\mahir\\AppData\\Local\\Temp\\pip-install-cjdf7ksc\\nglpy\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\mahir\AppData\Local\Temp\pip-wheel-vxwbreye'
       cwd: C:\Users\mahir\AppData\Local\Temp\pip-install-cjdf7ksc\nglpy\
  Complete output (36 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-3.9
  creating build\lib.win-amd64-3.9\nglpy
  copying nglpy\EmptyRegionGraph.py -> build\lib.win-amd64-3.9\nglpy
  copying nglpy\ngl.py -> build\lib.win-amd64-3.9\nglpy
  copying nglpy\PrebuiltGraph.py -> build\lib.win-amd64-3.9\nglpy
  copying nglpy\utils.py -> build\lib.win-amd64-3.9\nglpy
  copying nglpy\__init__.py -> build\lib.win-amd64-3.9\nglpy
  running build_ext
  building '_ngl' extension
  creating build\temp.win-amd64-3.9
  creating build\temp.win-amd64-3.9\Release
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.28.29910\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Ic:\users\mahir\appdata\local\programs\python\python39\include -Ic:\users\mahir\appdata\local\programs\python\python39\include -IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.28.29910\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt /EHsc /TpGraphStructure.cpp /Fobuild\temp.win-amd64-3.9\Release\GraphStructure.obj -std=c++11 -O3 -march=native
  cl: Є®¬ ­¤­ п бва®Є  warning D9002: Їа®ЇгбЄ ­ҐЁ§ўҐбв­®Ј® Ї а ¬Ґва  "-std=c++11"
  cl: Є®¬ ­¤­ п бва®Є  warning D9002: Їа®ЇгбЄ ­ҐЁ§ўҐбв­®Ј® Ї а ¬Ґва  "-O3"
  cl: Є®¬ ­¤­ п бва®Є  warning D9002: Їа®ЇгбЄ ­ҐЁ§ўҐбв­®Ј® Ї а ¬Ґва  "-march=native"
  GraphStructure.cpp
  C:\Users\mahir\AppData\Local\Temp\pip-install-cjdf7ksc\nglpy\GraphStructure.h(77): error C2061: бЁ­в ЄбЁзҐбЄ п ®иЁЎЄ : Ё¤Ґ­вЁдЁЄ в®а "string"
  C:\Users\mahir\AppData\Local\Temp\pip-install-cjdf7ksc\nglpy\GraphStructure.h(183): note: ўлЇ®«­пҐвбп Є®¬ЇЁ«пжЁп ббл«ЄЁ ­  нЄ§Ґ¬Ї«па Є« бб и Ў«®­ дг­ЄжЁЁ "GraphStructure<T>"
  C:\Users\mahir\AppData\Local\Temp\pip-install-cjdf7ksc\nglpy\GraphStructure.h(173): error C2061: бЁ­в ЄбЁзҐбЄ п ®иЁЎЄ : Ё¤Ґ­вЁдЁЄ в®а "string"
  GraphStructure.cpp(51): error C2244: GraphStructure<T>::compute_neighborhood: ­Ґ г¤ Ґвбп б®Ї®бв ўЁвм ®ЇаҐ¤Ґ«Ґ­ЁҐ дг­ЄжЁЁ бгйҐбвўгойҐ¬г ®Ўкпў«Ґ­Ёо
  GraphStructure.cpp(46): note:  б¬. ®Ўкпў«Ґ­ЁҐ "GraphStructure<T>::compute_neighborhood"
  GraphStructure.cpp(51): note: ®ЇаҐ¤Ґ«Ґ­ЁҐ
  GraphStructure.cpp(51): note: 'void GraphStructure<T>::compute_neighborhood(std::vector<int,std::allocator<int>> &,std::string,T,int &,bool)'
  GraphStructure.cpp(51): note: бгйҐбвўгойЁҐ ®Ўкпў«Ґ­Ёп
  GraphStructure.cpp(51): note: 'void GraphStructure<T>::compute_neighborhood(std::vector<int,std::allocator<int>> &)'
  GraphStructure.cpp(160): error C2244: GraphStructure<T>::GraphStructure: ­Ґ г¤ Ґвбп б®Ї®бв ўЁвм ®ЇаҐ¤Ґ«Ґ­ЁҐ дг­ЄжЁЁ бгйҐбвўгойҐ¬г ®Ўкпў«Ґ­Ёо
  GraphStructure.cpp(157): note:  б¬. ®Ўкпў«Ґ­ЁҐ "GraphStructure<T>::GraphStructure"
  GraphStructure.cpp(160): note: ®ЇаҐ¤Ґ«Ґ­ЁҐ
  GraphStructure.cpp(160): note: 'GraphStructure<T>::GraphStructure(std::vector<T,std::allocator<_Ty>> &,int,int,std::string,int,T,std::vector<int,std::allocator<int>> &,bool)'
  GraphStructure.cpp(160): note: бгйҐбвўгойЁҐ ®Ўкпў«Ґ­Ёп
  GraphStructure.cpp(160): note: 'GraphStructure<T>::GraphStructure(std::vector<T,std::allocator<_Ty>> &,int,int)'
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.28.29910\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2

  ERROR: Failed building wheel for nglpy
  Running setup.py clean for nglpy
Failed to build nglpy
Installing collected packages: nglpy
    Running setup.py install for nglpy ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\users\mahir\appdata\local\programs\python\python39\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\mahir\\AppData\\Local\\Temp\\pip-install-cjdf7ksc\\nglpy\\setup.py'"'"'; __file__='"'"'C:\\Users\\mahir\\AppData\\Local\\Temp\\pip-install-cjdf7ksc\\nglpy\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\mahir\AppData\Local\Temp\pip-record-dwnm8xiq\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\mahir\appdata\local\programs\python\python39\Include\nglpy'
         cwd: C:\Users\mahir\AppData\Local\Temp\pip-install-cjdf7ksc\nglpy\
    Complete output (36 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.9
    creating build\lib.win-amd64-3.9\nglpy
    copying nglpy\EmptyRegionGraph.py -> build\lib.win-amd64-3.9\nglpy
    copying nglpy\ngl.py -> build\lib.win-amd64-3.9\nglpy
    copying nglpy\PrebuiltGraph.py -> build\lib.win-amd64-3.9\nglpy
    copying nglpy\utils.py -> build\lib.win-amd64-3.9\nglpy
    copying nglpy\__init__.py -> build\lib.win-amd64-3.9\nglpy
    running build_ext
    building '_ngl' extension
    creating build\temp.win-amd64-3.9
    creating build\temp.win-amd64-3.9\Release
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.28.29910\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Ic:\users\mahir\appdata\local\programs\python\python39\include -Ic:\users\mahir\appdata\local\programs\python\python39\include -IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.28.29910\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt /EHsc /TpGraphStructure.cpp /Fobuild\temp.win-amd64-3.9\Release\GraphStructure.obj -std=c++11 -O3 -march=native
    cl: Є®¬ ­¤­ п бва®Є  warning D9002: Їа®ЇгбЄ ­ҐЁ§ўҐбв­®Ј® Ї а ¬Ґва  "-std=c++11"
    cl: Є®¬ ­¤­ п бва®Є  warning D9002: Їа®ЇгбЄ ­ҐЁ§ўҐбв­®Ј® Ї а ¬Ґва  "-O3"
    cl: Є®¬ ­¤­ п бва®Є  warning D9002: Їа®ЇгбЄ ­ҐЁ§ўҐбв­®Ј® Ї а ¬Ґва  "-march=native"
    GraphStructure.cpp
    C:\Users\mahir\AppData\Local\Temp\pip-install-cjdf7ksc\nglpy\GraphStructure.h(77): error C2061: бЁ­в ЄбЁзҐбЄ п ®иЁЎЄ : Ё¤Ґ­вЁдЁЄ в®а "string"
    C:\Users\mahir\AppData\Local\Temp\pip-install-cjdf7ksc\nglpy\GraphStructure.h(183): note: ўлЇ®«­пҐвбп Є®¬ЇЁ«пжЁп ббл«ЄЁ ­  нЄ§Ґ¬Ї«па Є« бб и Ў«®­ дг­ЄжЁЁ "GraphStructure<T>"
    C:\Users\mahir\AppData\Local\Temp\pip-install-cjdf7ksc\nglpy\GraphStructure.h(173): error C2061: бЁ­в ЄбЁзҐбЄ п ®иЁЎЄ : Ё¤Ґ­вЁдЁЄ в®а "string"
    GraphStructure.cpp(51): error C2244: GraphStructure<T>::compute_neighborhood: ­Ґ г¤ Ґвбп б®Ї®бв ўЁвм ®ЇаҐ¤Ґ«Ґ­ЁҐ дг­ЄжЁЁ бгйҐбвўгойҐ¬г ®Ўкпў«Ґ­Ёо
    GraphStructure.cpp(46): note:  б¬. ®Ўкпў«Ґ­ЁҐ "GraphStructure<T>::compute_neighborhood"
    GraphStructure.cpp(51): note: ®ЇаҐ¤Ґ«Ґ­ЁҐ
    GraphStructure.cpp(51): note: 'void GraphStructure<T>::compute_neighborhood(std::vector<int,std::allocator<int>> &,std::string,T,int &,bool)'
    GraphStructure.cpp(51): note: бгйҐбвўгойЁҐ ®Ўкпў«Ґ­Ёп
    GraphStructure.cpp(51): note: 'void GraphStructure<T>::compute_neighborhood(std::vector<int,std::allocator<int>> &)'
    GraphStructure.cpp(160): error C2244: GraphStructure<T>::GraphStructure: ­Ґ г¤ Ґвбп б®Ї®бв ўЁвм ®ЇаҐ¤Ґ«Ґ­ЁҐ дг­ЄжЁЁ бгйҐбвўгойҐ¬г ®Ўкпў«Ґ­Ёо
    GraphStructure.cpp(157): note:  б¬. ®Ўкпў«Ґ­ЁҐ "GraphStructure<T>::GraphStructure"
    GraphStructure.cpp(160): note: ®ЇаҐ¤Ґ«Ґ­ЁҐ
    GraphStructure.cpp(160): note: 'GraphStructure<T>::GraphStructure(std::vector<T,std::allocator<_Ty>> &,int,int,std::string,int,T,std::vector<int,std::allocator<int>> &,bool)'
    GraphStructure.cpp(160): note: бгйҐбвўгойЁҐ ®Ўкпў«Ґ­Ёп
    GraphStructure.cpp(160): note: 'GraphStructure<T>::GraphStructure(std::vector<T,std::allocator<_Ty>> &,int,int)'
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.28.29910\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\users\mahir\appdata\local\programs\python\python39\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\mahir\\AppData\\Local\\Temp\\pip-install-cjdf7ksc\\nglpy\\setup.py'"'"'; __file__='"'"'C:\\Users\\mahir\\AppData\\Local\\Temp\\pip-install-cjdf7ksc\\nglpy\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\mahir\AppData\Local\Temp\pip-record-dwnm8xiq\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\mahir\appdata\local\programs\python\python39\Include\nglpy' Check the logs for full command output.`

In the code above warning D9002 is incorrect encoding for "skipping unknown parameter". I have googled it and it turns out that MSVS Build Tools does not support C++11 which is likely used for NGL library you wrapped up in python. The errors during the compilation are also incorrectly encoded (likely C1251 as UTF-8) but they state: unknown variable etc. Likely caused by the absence of C++11.

How should I install your package?

**Comment by dmaljovec** [comment]
Hmm, good question. It has been awhile since I have looked at this, but that flag is enabled [here](https://github.com/maljovec/nglpy/blob/master/setup.py#L97). I'll double check the code to see if it is actually needed or how hard it would be to modify it without that.

Meanwhile it does look like Microsoft has been working on [this](https://devblogs.microsoft.com/cppblog/c11-and-c17-standard-support-arriving-in-msvc/). Odd that this post is for an older version yet it still does not work for you. I didn't read it very closely, but maybe there are some toggles that need to be switched to get C++11 support? :shrug:

I have not encountered this problem before, but do have a Windows machine lying around somewhere where I can try this out. I am a bit swamped at my day job, so it might be some time before I can look more closely. If the above is not helpful, Is Windows Subsystem Linux an option for you while I dig into this more in the meanwhile? Alternatively, is installing this alongside whatever else you are building/using in a Linux-based Docker image a possibility? I apologize these are not fixes, but instead workarounds, but I'd am hoping it buys me some time to test things on my own Windows machine and see if I can replicate or not.

**Comment by dmaljovec** [comment]
FWIW, I [tested out](https://github.com/maljovec/nglpy/tree/build-no-c+%2B11) removing the references to C++11 in the `setup.py` file and tried to do an install on a virtual environment on my Mac and it appears to have worked. You could try cloning the repo and pulling the `build-no-c++11` branch. You might need to install [swig](http://www.swig.org/) to compile from source. If you can do a `make all && pip install -e .` and it works for you, then perhaps I will make that part of the mainline build. Requiring less from users seems better, and I am not sure why I enabled that compiler flag in the first place to be honest. 🤔

**Comment by Antillar** [comment]
Ok, so, first, I installed swig and added it to the Path variable.
Next, I downloaded your branch and tried to build it.
Upon execution of the command you provided I get the following:
![image](https://user-images.githubusercontent.com/12700019/117687969-0247da80-b1c1-11eb-92fa-c5ce83b14952.png)
The error states (when translated from Russian) - file not found.
As you can see, there is no ngl.py in the repo. Should it be there? Should it appear after the execution of the first command?

**Comment by Antillar** [comment]
When trying to compile straight from git with pip install 
`pip install git+https://github.com/maljovec/nglpy.git@build-no-c++11#egg=nglpy` 
I get these errors when building wheel for nglpy:
![image](https://user-images.githubusercontent.com/12700019/117689491-7931a300-b1c2-11eb-92da-5f344dd649ec.png)
Errors from this screenshot, correctly decoded with [online decoder](https://2cyr.com/decode/?lang=ru) and translated to English:
GraphStructure.h(77): syntax error: identifier "string"
GraphStructure.h(173): same as above
GraphStructure.cpp(51): GraphStructure<T>::compute_neighborhood call does not match definition
GraphStructure.cpp(160): same as above
`void compute_neighborhood(std::vector<int> &edgeIndices,
                            std::string type, T beta, int &kmax,
                            bool connect = false);`
Does my compiler not know what std::string is?

Edit: found out that this may be the case if `#include<string>` is missing from the header file. It is not included there. Can you please check this, maybe include it and see what happens?

**Comment by Antillar** [comment]
I forked your repo and added `#include<string>` in GraphStructure.h
Now when I do `pip install git+https://github.com/Antillar/nglpy.git@build-no-c++11#egg=nglpy`
I first get a bunch of warnings about 
 ```
with
          [
              T=double
          ]
```
but not a single error. Afterwards it produces the following:
 ```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30031\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Ic:\users\mahir\appdata\local\programs\python\python39\include -Ic:\users\mahir\appdata\local\programs\python\python39\include -IC:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30031\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30031\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.20344.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.20344.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.20344.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.20344.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.20344.0\cppwinrt /EHsc /TpUnionFind.cpp /Fobuild\temp.win-amd64-3.9\Release\UnionFind.obj -O3 -march=native
  cl: Є®¬ ­¤­ п бва®Є  warning D9002: Їа®ЇгбЄ ­ҐЁ§ўҐбв­®Ј® Ї а ¬Ґва  "-O3"
  cl: Є®¬ ­¤­ п бва®Є  warning D9002: Їа®ЇгбЄ ­ҐЁ§ўҐбв­®Ј® Ї а ¬Ґва  "-march=native"
  UnionFind.cpp
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30031\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Ic:\users\mahir\appdata\local\programs\python\python39\include -Ic:\users\mahir\appdata\local\programs\python\python39\include -IC:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30031\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.29.30031\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.20344.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.20344.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.20344.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.20344.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.20344.0\cppwinrt /EHsc /Tpngl_wrap.cpp /Fobuild\temp.win-amd64-3.9\Release\ngl_wrap.obj -O3 -march=native
  cl: Є®¬ ­¤­ п бва®Є  warning D9002: Їа®ЇгбЄ ­ҐЁ§ўҐбв­®Ј® Ї а ¬Ґва  "-O3"
  cl: Є®¬ ­¤­ п бва®Є  warning D9002: Їа®ЇгбЄ ­ҐЁ§ўҐбв­®Ј® Ї а ¬Ґва  "-march=native"
  ngl_wrap.cpp
  c1xx: fatal error C1083: ЌҐ г¤ Ґвбп ®вЄалвм д ©« Ёбв®з­ЁЄ: ngl_wrap.cpp: No such file or directory,
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Preview\\VC\\Tools\\MSVC\\14.29.30031\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
```
As you see here, it does not find the file ngl_wrap.cpp somehow. I have downloaded your repo and know it is there. So idk where the problem lies.

**Comment by dmaljovec** [comment]
it has been awhile, but I have a branch where I was working on converting this to be an entirely pythonic project: https://github.com/maljovec/nglpy/pull/10. You could try building that? It should be simpler like just checking out the branch and doing `pip install -e .` from the project root, maybe? It adds a new experimental feature, but I think the core functionality and interface should remain largely the same for you. How big of data are you looking to explore?

**Comment by dmaljovec** [comment]
> Edit: found out that this may be the case if #include<string> is missing from the header file. It is not included there. Can you please check this, maybe include it and see what happens?

There is no reason not to include this in the current iteration that I can think of. I'd be happy to make this change to the current code that does depend on the C++ code.

**Comment by dmaljovec** [comment]
> As you can see, there is no ngl.py in the repo. Should it be there? Should it appear after the execution of the first command?

Yes and yes. The swig command should generate `ngl.py` and the `ngl_wrap.cpp` file.

**Comment by Antillar** [comment]
> it has been awhile, but I have a branch where I was working on converting this to be an entirely pythonic project: #10. You could try building that?

I tried it before and it also fails on building wheel. I will try the local installation now.

I can also check the installation on Windows Server 2008 now, will try this in a few days, maybe it is version specific bug

**Comment by Antillar** [comment]
Got these errors (the same ones) when asking a friend to install the library on Windows 10:
![image](https://user-images.githubusercontent.com/12700019/138661415-5335f265-6191-4623-85c5-2e1c227d4d0a.png)
The issue persists as you can see.
Getting from the branch you have made for me also did not work.
Meanwhile on Linux it works just fine for me:
![image](https://user-images.githubusercontent.com/12700019/138661635-c0edae73-2029-4013-abaa-f7d9356acdeb.png)
So, well, there is definitely an issue with your library on Windows 10. How to solve it? no idea. 
But I got what I wanted with a virtualbox workaround.

**Comment by dmaljovec** [comment]
It looks like Microsoft is suggesting to install Microsoft C++ build tools 14.0 or greater. Did you try that on the Windows machine?

**Comment by Antillar** [comment]
I have it installed, yes. Nothing changes--Sent from Yandex.Mail for mobile27.10.2021, 15:42, "Dan Maljovec" ***@***.***>:
It looks like Microsoft is suggesting to install Microsoft C++ build tools 14.0 or greater. Did you try that on the Windows machine?

**Comment by dmaljovec** [comment]
I just recently had a similar [install error](https://github.com/maljovec/nglpy/issues/27) on Linux. I'll try to set up the CI system to publish wheels for Windows as well.

**Comment by dmaljovec** [comment]
Hopefully, that will obviate the need for any compilation on your end.

**Comment by maljovec** [comment]
I apologize this is clearly a day late and a dollar short to be useful for you, but I finally put aside some time to auto-publish wheels for this package. You should now be able to install a binary distribution using one of the wheels published with the recently minted version 1.1.3: https://pypi.org/project/nglpy/#files. Again, I apologize for the year long delay.

**Comment by Antillar** [comment]
Hi!
I have graduated and no longer have the environment required to test your package available. 
I am glad it works now, but I am no longer able to try it out. Thank you.
